cmd/object-handlers.go

```diff
@@ -576,7 +573,7 @@ func (api objectAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Req
 			sha256sum = r.Header.Get("X-Amz-Content-Sha256")
 		}
 		// Create object.
-		objInfo, err = objectAPI.PutObject(bucket, object, NewHashReader(r.Body, size, metadata["etag"], sha256sum), metadata)
+		objInfo, err = objectAPI.PutObject(bucket, object, NewHashReader(r.Body, size, md5sum, sha256sum), metadata)
```

Comment:
These changes are just diff increaser.  You can optimize things is different PR IMO

Reply:
This is not unrelated @balamurugana

Reply:
No.  In the above `md5sum = metadata["etag"]`. you are just optimizing map access.

Reply:
no there is no such thing as "etag" saved in `metadata` it is only saved internally at ObjectLayer. There is no reason to add it prematurely.  You have to see `md5sum` is never set to a map. Also taking "etag" as the md5sum value is incorrect from readability point of view.